src/Compilers/CSharp/Test/Emit/CodeGen/StackAllocInitializerTests.cs

```diff
@@ -9,6 +9,46 @@ namespace Microsoft.CodeAnalysis.CSharp.UnitTests
     [CompilerTrait(CompilerFeature.StackAllocInitializer)]
     public class StackAllocInitializerTests : CompilingTestBase
     {
+        [Fact]
+        public void TestElementThrow()
+        {
+            var text = @"
+using System;
+
+static unsafe class C
+{
+    static void Use(int* i) {}
+    
+    static int M() { return 0; }
+
+    static void Main()
+    {
+        var p = stackalloc[] { M(), true ? throw null : M(), M() };
+        Use(p);
```

Comment:
The "no assignment while evaluating" could be tested with code like:

```C#
using System;
public unsafe class C {
    
    static byte* ptr;
    
    public static void Main() 
    {
        ptr = stackalloc byte[3];
        
        try
        {
        	Test();
        }
        catch
        {
        }
        
        // should print 1
        System.Console.WriteLine(ptr[0]);
    }

    public static void Test()
    {
        // make sure that while evaluating the elements, 
        // the ptr is not yet assigned a new value.
        ptr = stackalloc byte[3]
                            {
                                4, 
                                GetAndPrint(ptr[1]) /* should be 2 */, 
                                Throw()
                            };   
    }                                         
                                     
    public static byte Throw()
    {
        throw new Exception();
    }    
                                     
    public static byte GetAndPrint(byte b)
    {
        return b;
    }    
}
```

Reply:
I am not sure the same pattern is possible with Span. I think escape rules should not allow it. Then we don't care about that case :-)

Reply:
Ignore the above. That is not possible even with pointers. The reason is that the target variable must be a local.  
We can store in local first and then to a field, but that is not interesting.

Reply:
tried a couple of variations with no luck. good to know it wasn't possible 😄